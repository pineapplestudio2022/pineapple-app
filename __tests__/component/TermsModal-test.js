import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {render} from '@testing-library/react-native';
import TermsModal from '../../src/Components/TermsandConditionModal';

const wrapper = ({children}) => (
  <NativeBaseProvider
    initialWindowMetrics={{
      frame: {x: 0, y: 0, width: 0, height: 0},
      insets: {top: 0, left: 0, right: 0, bottom: 0},
    }}>
    {children}
  </NativeBaseProvider>
);

let props;
let component;

const getTempComponent = props => {
  return <TermsModal {...props} />;
};

describe('[TermsModal] Test', () => {
  props = {terms: 1};
  component = getTempComponent(props);
  test('TermsModal snapshot', () => {
    jest.useFakeTimers();
    const rendered = render(component, {wrapper});
    expect(rendered).toMatchSnapshot(); //기존 스냅샷과 일치하는지
    expect(rendered).toBeTruthy(); //컴포넌트가 null, undefinded 같이 falsy 한 값을 가지지 않았는지
  });
});
