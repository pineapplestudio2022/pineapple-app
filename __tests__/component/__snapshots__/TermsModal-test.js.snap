// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[TermsModal] Test TermsModal snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 37.97864768683274,
        "opacity": 1,
        "width": 46.15384615384615,
      }
    }
  >
    <Image
      accessibilityLabel=" "
      alt=" "
      maxWidth="100%"
      onError={[Function]}
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/Assets/Image/member/icon_signup_externallink_gray.png",
        }
      }
      style={
        Array [
          Object {
            "maxWidth": "100%",
          },
          Object {
            "width": "100%",
          },
        ]
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
